pkg/util/validation/validation.go

```diff
@@ -247,19 +247,24 @@ func IsHTTPHeaderName(value string) []string {
 	return nil
 }
 
-const configMapKeyFmt = "\\.?" + dns1123SubdomainFmt
+const configMapKeyFmt = `[-._a-zA-Z0-9]+`
 
 var configMapKeyRegexp = regexp.MustCompile("^" + configMapKeyFmt + "$")
 
-// IsConfigMapKey tests for a string that conforms to the definition of a
-// subdomain in DNS (RFC 1123), except that a leading dot is allowed
+// IsConfigMapKey tests for a string that is a valid key for a ConfigMap or Secret
 func IsConfigMapKey(value string) []string {
 	var errs []string
 	if len(value) > DNS1123SubdomainMaxLength {
 		errs = append(errs, MaxLenError(DNS1123SubdomainMaxLength))
 	}
 	if !configMapKeyRegexp.MatchString(value) {
-		errs = append(errs, RegexError(configMapKeyFmt, "key.name"))
+		errs = append(errs, RegexError(configMapKeyFmt, "key.name", "KEY_NAME", "key-name"))
+	}
+	if value == "." {
+		errs = append(errs, `must not be "."`)
+	}
+	if strings.HasPrefix(value, "..") {
+		errs = append(errs, `must not be prefixed with ".."`)
```

Comment:
I know this catches it, but I think it would be clearer to test for == ".." like you did for "." and return a more precise error.  Of course, both that and this will trigger, so this should be the else.

```
if value == ".." {
	errs = append(errs, `must not be '..'`)
} else if strings.HasPrefix(value, "..") {